docs/quickstarts/nextjs.mdx

```diff
@@ -42,7 +42,13 @@ CLERK_SECRET_KEY={{secret}}
 
 ### Mount `<ClerkProvider />`
 
-Update your root layout to include the [`<ClerkProvider />`](/docs/components/clerk-provider#clerk-provider) wrapper. The `<ClerkProvider />` component wraps your Next.js application to provide active session and user context to Clerk's hooks and other components. It is recommended that the `<ClerkProvider />` wraps the `<body/>` to enable the context to be accessible anywhere within the app.
+Add the [`<ClerkProvider />`](/docs/components/clerk-provider#clerk-provider) wrapper to your application. The `<ClerkProvider />` component wraps your Next.js application to provide active session and user context to Clerk's hooks and other components. A reasonable default approach is to have `<ClerkProvider />` wrap the `<body/>` to enable the context to be accessible anywhere within the app.
+
+<Callout type="info">
+  The `<ClerkProvider />` component needs to access [headers](https://nextjs.org/docs/app/api-reference/functions/headers) to authenticate correctly. This means anything wrapped by the provider will be opted into [dynamic rendering](https://nextjs.org/docs/app/building-your-application/rendering/server-components#dynamic-rendering) at request time. If you have static-optimized or ISR pages that you would prefer not to be opted into dynamic rendering, make sure they are <em>not</em> wrapped by `<ClerkProvider />`.
+  
+  This is easiest to accomplish by ensuring that `<ClerkProvider />` is added further down the tree to wrap page groups that explicitly need authentication instead of having the provider wrap your application root as recommended above. For example, if you have a set of landing/marketing pages, and an app that requires login that live within the same project, wrapping just the app layout group with the provider and keeping the marketing pages in a separate layout group would allow the marketing pages to continue to be static-optimized.

```

Comment:
Tweaked the wording a bit:
```suggestion
  This is easiest to accomplish by ensuring that `<ClerkProvider />` is added further down the tree to wrap route groups that explicitly need authentication instead of having the provider wrap your application root as recommended above. For example, if you have a set of landing/marketing pages and a dashboard that requires login within the same project, create separate `(marketing)` and `(dashboard)` route groups. Rendering `<ClerkProvider />` in the `(dashboard)/layout.jsx` layout file will ensure that only those routes are opted into dynamic rendering, allowing the marketing routes to be statically optimized.
```

Reply:
Great call, will apply this now 🙏